### PR TITLE
basic shell and test for bootstrapping transformers from a pubspec

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

**Comment by jakemac53** [comment]
Note that this doesn't actually test functionality at all, and doesn't run the transformers anyways. But, its a start :).

File: lib/src/bazelify/pubspec.dart [hunk 1]
```diff
@@ -96,10 +96,22 @@ class Transformer {
   /// The name of the transformer as it appears in the pubspec.
   final String name;
 
-  /// The config supplied to the transformer, may be null.
+  /// The config supplied to the transformer.
   final Map config;
 
-  Transformer(this.name, {this.config});
+  /// The uri for this transformer.
```
Comment (natebosch) [inline lib/src/bazelify/pubspec.dart]: IMO this doc isn't adding any new information. Can it be expanded or removed?

File: lib/src/bazelify/transformers.dart [hunk 2]
```diff
@@ -2,24 +2,54 @@
...
 /// The mirror system.
 ///
 /// Cached to avoid re-instantiating each time a transformer is initialized.
 final _mirrors = currentMirrorSystem();
 
+/// Creates the contents of a library which exposes a `buildTransformers` method
+/// that you can call to get concrete transformer phases from a `Pubspec`.
+String bootstrapTransformersFromPubpec(Pubspec pubspec) {
+  var sb = new StringBuffer();
+  sb.writeln('import "dart:convert";');
```
Comment (natebosch) [inline lib/src/bazelify/transformers.dart]: how about .. cascades?

File: test/projects/simple_with_transformer/lib/transformer.dart [hunk 3]
```diff
@@ -3,8 +3,40 @@ import 'package:barback/barback.dart';
...
+class MyTransformerGroup implements TransformerGroup {
+  final BarbackSettings settings;
+
+  MyTransformerGroup(this.settings);
+  MyTransformerGroup.asPlugin(this.settings);
+
+  @override
+  Iterable<Iterable> get phases => [
+    [
+      new MyTransformer(settings),
```
Comment (natebosch) [inline test/projects/simple_with_transformer/lib/transformer.dart]: [nit] might format more compactly without the trailing commas

**Review by natebosch — APPROVED** [review]
When you merge it would be useful to add more detail to the commit message than what's in the pull request

**Comment by jakemac53** [comment]
Closing after offline discussion, packages that expose a transformer which they also want to work in bazel will need to add some extra config. At the very least we need to know the outputs ahead of time so something is required anyways, and it can greatly simplify things.

**Pull request closed without merging**